### PR TITLE
JAMES-3603 Get rid of some top level folders in project hierarchy

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Review by jeantil — APPROVED** [review]

**Comment by jeantil** [comment]
thank you infinitely for make small focused commits, this made reviewing your changes very fast !

**Review by Arsnael — COMMENTED** [review]
I think the links in the root README in the section `How to run James in Docker` are broken too and should be fixed :)

File: server/protocols/jmap-draft-integration-testing/memory-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/memory/MemorySpamAssassinContractTest.java [hunk 1]
```diff
@@ -1,82 +0,0 @@
...
-import org.apache.james.MemoryJamesServerMain;
-import org.apache.james.jmap.draft.methods.integration.SpamAssassinContract;
-import org.apache.james.jmap.draft.methods.integration.SpamAssassinModuleExtension;
-import org.apache.james.junit.categories.Unstable;
-import org.apache.james.modules.TestJMAPServerModule;
-import org.apache.james.spamassassin.SpamAssassinExtension;
-import org.junit.jupiter.api.Tag;
-import org.junit.jupiter.api.extension.RegisterExtension;
-
-class MemorySpamAssassinContractTest implements SpamAssassinContract {
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply server/protocols/jmap-draft-integration-testing/memory-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/memory/MemorySpamAssassinContractTest.java:33]: Mistake of course

**Comment by chibenwa** [comment]
Force pushed to re-add the missing file (first commit).

Added the last commit to fix README links as suggested by @Arsnael

**Review by Arsnael — APPROVED** [review]

**Pull request merged**